### PR TITLE
fix and update poetry dependencies

### LINKED ISSUE #393: Issue with poetry build-system section when including as a dependency
When including the GMT as a dependency of another project, the `poetry install` command on the other project results in an error that this project's `pyproject.toml` file is incorrect.

```
  Command ['/Users/nlong/.pyenv/versions/3.7.9/envs/3.7.9-gmt-ex/bin/pip', 'install', '--no-deps', '-U', '/Users/nlong/working/urban-analysis/geojson-modelica-translator'] errored with the following return code 1, and output:
  Processing /Users/nlong/working/urban-analysis/geojson-modelica-translator
    DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
     pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
  ERROR: file:///Users/nlong/working/urban-analysis/geojson-modelica-translator has a pyproject.toml file that does not comply with PEP 518: 'build-system.requires' contains an invalid requirement: 'poetry = 1.1.6'


  at ~/.pyenv/versions/3.7.9/envs/3.7.9-gmt-ex/lib/python3.7/site-packages/poetry/utils/env.py:1075 in _run
      1071│                 output = subprocess.check_output(
      1072│                     cmd, stderr=subprocess.STDOUT, **kwargs
      1073│                 )
      1074│         except CalledProcessError as e:
    → 1075│             raise EnvCommandError(e, input=input_)
      1076│
      1077│         return decode(output)
      1078│
      1079│     def execute(self, bin, *args, **kwargs):
```

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?
We were using an older pyproject.toml section for `[build-system]`. This was causing an issue when requiring the package from another project.

#### What does this PR accomplish?
New format is the following.
```
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

#### How should this be manually tested?
* make sure everything runs (`poetry update`, `poetry run py.test`)

#### What are the relevant tickets?
Fixes #393

### CODE REVIEW

**Comment by nllong** [comment]
welp, this didn't solve it @macintoshpie. I thought it worked because it fixed the installation from the GMT-examples side, but then broke just installing the GMT by itself. ¯\_(ツ)_/¯

**Comment by macintoshpie** [comment]
@nllong looks like there was a small typo in the pyproject.toml, but should be fixed now and works with the examples repo. If this looks good to you then I think we can merge this

File: pyproject.toml [hunk 1]
```diff
@@ -53,6 +53,7 @@ sphinx-jsonschema = "1.16.7"
 toml = "0.10.2"
 tox = "3.20.0"
 
+[build-system]
```
Comment (nllong) [inline pyproject.toml:56]: 😲 😔

**Review by macintoshpie — APPROVED** [review]
👍

**Pull request merged**